src/clients/BridgeClient.ts

```diff
@@ -61,6 +63,30 @@ export class BridgeClient implements IClient {
                 this._logger.trace(TelemetryEvent.BridgeClient_GetVersionSuccess);
                 return bridgeVersion;
             };
+            const codesignAsyncFn = async (): Promise<string> => {
+                const args: string[] = [ `-s`, `-`, this._executablePath ];
+                return await this._commandRunner.runAsync(
+                    this.codesignPath,
+                    args,
+                    null, // currentWorkingDirectory
+                    null  // customEnvironmentVariables
+                );
+            };
+
+            if (process.platform === `darwin` 
+                && process.arch === `arm64` 
+                && await fileSystem.existsAsync(this.codesignPath)) {
+                try {
+                    const output = await RetryUtility.retryAsync<string>(codesignAsyncFn, 1, 100);
+                    if (output.includes(`is already signed`)) {
+                        return; //codesigning is already done
+                    }
+                } catch (error) {
+                    if (!error.message.includes(`is already signed`)) {
```

Comment:
💡 Thinking out loud for my own understanding: So in case `error` happens, and `then` we check if it is not `signed` then bubble the error. What will happen, if `error` happens and it is `signed`?

Reply:
if the binary is already signed it will be executable and user wouldn't face any issues.